### PR TITLE
feat: Add Python 3.11 to the testing

### PR DESCRIPTION
**Pull request by cclauss**
# Fixes #

A short description of what this PR does.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/sendgrid/sendgrid-python/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://support.sendgrid.com).

### CODE REVIEW

**Comment by sk-** [comment]
@cclauss do you have plans to continue this PR?

It would also be good to add Python 3.11 to the list of [classifiers](https://github.com/sendgrid/sendgrid-python/blob/4288aa618b04a3e99173f59ce43d7a4b379b93c4/setup.py#L43).

The CI is failing because the name of the PR is not following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) (this is the [action](https://github.com/amannn/action-semantic-pull-request) checking that).

**Comment by cclauss** [comment]
I was worried to touch the classifiers because `python_requires` says no Py34 and the `classifiers` say yes Py34 when in fact no one uses Py34 anymore.  https://devguide.python.org/versions

**Comment by sk-** [comment]
But we are still waiting for an approval from maintainers to be able to run the tests. @childish-sambino could you grant permissions.

Screen Shot 2023-03-11 at 15 01 14

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**
